Sprint-1/stretch/aoc-2018-day1/solution.js

```diff
@@ -0,0 +1,44 @@
+const fs = require("fs");
+const path = require("path");
+
+// Get the current directory of the script
+const filePath = path.join(__dirname, "./input.txt");
+
+fs.readFile(filePath, "utf8", (err, data) => {
+  if (err) {
+    console.error("Error reading file:", err);
+    return;
+  }
+
+  const changes = data.split("\n").filter((line) => line.trim() !== "");
+
+  // Part One: Calculate the final frequency
+  const finalFrequency = calculateFrequency(changes);
+  console.log("Final frequency after all changes:", finalFrequency);
+
+  // Part Two: Find the first repeated frequency
+  const firstRepeatedFrequency = findFirstRepeatedFrequency(changes);
+  console.log("First repeated frequency:", firstRepeatedFrequency);
+});
+
+function calculateFrequency(changes) {
+  //   let frequency = 0;
+  //   for (const change of changes) {
+  //     frequency += parseInt(change, 10);
+  //   }
+  return changes.reduce((acc, change) => acc + Number(change), 0);
+}
+function findFirstRepeatedFrequency(changes) {
+  let frequency = 0;
+  const seenFrequencies = new Set([frequency]);
```

Comment:
Why start with a set with one value, 0, instead of an empty set?

Reply:
What is the spec of step 2? I couldn't find them.

Reply:
![Screenshot 2024-11-30 at 12 39 53](https://github.com/user-attachments/assets/6f6e7cd8-7a6f-4c9a-ae1e-3ee94c6a52d2)

Reply:
Sorry. I meant Part 2. The spec for `findFirstRepeatedFrequency()`.

Reply:
Sorry, Now I don’t understand 
https://adventofcode.com/2018/day/1